File: 2021-09/행렬의덧셈.js
function solution(arr1, arr2) {
    var answer = [];
    const I_SIZE = arr1.length;
    const J_SIZE = arr1[0].length;
    for(let i = 0; i < I_SIZE; i++) {
        answer[i] = [];
        for(let j = 0; j < J_SIZE; j++) {
            answer[i].push(arr1[i][j] + arr2[i][j]);
        }
    }
    return answer;
}

console.log(solution([[1, 2], [2,3]], [[3, 4], [5, 6]])); //[[4, 6], [7, 9]]

/* 
    다른사람풀이 배운 코드
    새로 만들지 않고, A에 더한다는 생각을 못했었다.
*/

function solution(A, B) {
    return A.map((a,i) => a.map((b, j) => b + B[i][j]));
}

File: programmers/src/main/java/level1/행렬의덧셈.java
package level1;

public class 행렬의덧셈 {

	public int[][] solution(int[][] arr1, int[][] arr2) {
		int size = arr1.length;
		int size2 = arr1[0].length;
		int[][] answer = new int[size][size2];

		for (int i = 0; i < size; i++) {
			for (int j = 0; j < size2; j++) {
				answer[i][j] = arr1[i][j] + arr2[i][j];
			}
		}

		return answer;
	}

}
